crates/node-db/src/lib.rs

```diff
@@ -233,15 +217,63 @@ pub fn delete_state(
 }
 
 /// Fetches a solution by its content address.
-pub fn get_solution(
-    conn: &Connection,
-    ca: &ContentAddress,
-) -> Result<Option<Solution>, QueryError> {
-    let mut stmt = conn.prepare(sql::query::GET_SOLUTION)?;
-    let solution_blob: Option<Vec<u8>> = stmt
-        .query_row([ca.0], |row| row.get("solution"))
-        .optional()?;
-    Ok(solution_blob.as_deref().map(decode).transpose()?)
+pub fn get_solution(tx: &Transaction, ca: &ContentAddress) -> Result<Solution, QueryError> {
+    let mut data_stmt = tx.prepare(sql::query::GET_SOLUTION_DATA)?;
+    let mut data = data_stmt
+        .query_map([ca.0], |row| {
+            let contract_addr = row.get::<_, Hash>("contract_addr")?;
+            let predicate_addr = row.get::<_, Hash>("predicate_addr")?;
+            Ok(SolutionData {
+                predicate_to_solve: PredicateAddress {
+                    contract: ContentAddress(contract_addr),
+                    predicate: ContentAddress(predicate_addr),
+                },
+                state_mutations: vec![],
+                decision_variables: vec![],
+            })
+        })?
+        .collect::<Result<Vec<_>, _>>()?;
+    data_stmt.finalize()?;
+
+    let mut dec_vars_stmt = tx.prepare(sql::query::GET_SOLUTION_DEC_VARS)?;
+    let mut mutations_stmt = tx.prepare(sql::query::GET_SOLUTION_MUTATIONS)?;
+
+    for (data_ix, datum) in data.iter_mut().enumerate() {
+        let mut mutations = vec![];
+        let mut dec_vars = vec![];
+
+        // Fetch the mutations.
+        let mut mutation_rows = mutations_stmt.query(named_params! {
+            ":content_hash": ca.0,
+            ":data_index": data_ix,
+        })?;
+        while let Some(mutation_row) = mutation_rows.next()? {
+            let key_blob: Vec<u8> = mutation_row.get("key")?;
+            let value_blob: Vec<u8> = mutation_row.get("value")?;
+            let key: Key = words_from_blob(&key_blob);
+            let value: Value = words_from_blob(&value_blob);
+            mutations.push(Mutation { key, value });
+        }
+
+        // Fetch the decision variables.
+        let mut dec_var_rows = dec_vars_stmt.query(named_params! {
+            ":content_hash": ca.0,
+            ":data_index": data_ix,
+        })?;
+        while let Some(dec_var_row) = dec_var_rows.next()? {
+            let value_blob: Vec<u8> = dec_var_row.get("value")?;
+            let value: Value = words_from_blob(&value_blob);
+            dec_vars.push(value);
```

Comment:
This is relying on the order these were inserted in the db which isn't always the same.

Reply:
All three of solution data, dec vars and mutations are ordered by ascending index